### ISSUE TITLE
Request: Publish a json schema for use with semaphore YAML files

### ISSUE THREAD

**Issue by bbugh** [semaphoreci/docs#1230]
schemastore.org is tracking JSON schemas. RedHat's yaml-language-server supports JSON schema parsing for YAML files pulled from schemastore.org. Many systems are using it now: GitHub's actions and workflows, graphql-code-generator's codegen.yml, CircleCI's config.yml, etc. 

It's nice to have a linter in the CLI, but it would be very helpful for self-documenting files and validation if the semaphore.yml schema was published to schemastore.org, which tools (including yaml-language-server) could use to validate. There's lots of different ways to generate them.
 
![schemaphore](https://user-images.githubusercontent.com/438465/186147399-7ee2fcba-d891-4607-a461-6d0dbea218f4.png)

**Comment by pieterocp** [comment]
Yeah, this is a _big_ qol upgrade since you can get much faster feedback loops over having to push repeatedly when fiddling with config. It should only take about a week or so to get a decent working version together.

Given there is [a 41:1 ROI ](https://semaphoreci.com/wp-content/uploads/2020/03/Semaphore_ROI.pdf) using semaphore, you can claim even more with speedier feedback.

**Comment by pieterocp** [comment]
Until this is done, I've had to slap a docblock at the top (like below) so it doesn't think it's a appveyor file and suggests bad changes.

```yaml
# yaml-language-server: $schema=https://json.schemastore.org/yamllint.json
```

**Comment by pieterocp** [comment]
Since the big migration to open source (see https://github.com/semaphoreio/semaphore/issues/28 for this being a suggestion when editing via the workflow editor), it might be a lot easier to start to implement this?
